kernel/parser.h

```diff
@@ -3,17 +3,30 @@
 
 #include <stddef.h>
 
+/// This container stores a string split up by whitespace
```

Comment:
what about just shortening the struct explanation into a single line that roughly explains what is it
and then make a comment after every variable decleration in a struct?
that seems like a better option in my opinion